contracts/Removal.sol

```diff
@@ -33,18 +34,27 @@ contract Removal is
     bool list;
   }
 
-  uint256 private _tokenIdCounter;
+  /**
+   * @notice The EscrowedNORI contract that manages escrowed tokens.
+   */
+  EscrowedNORI private _eNORI;
+  uint256 public tokenIdCounter;
```

Comment:
I think I've personally had a change of hear on whether we should expose any state variables as public. I think we should make this private in a base contract and instead expose a virtual getter/setter function to make it obvious that the only way this variable interaction should be changed is via an override. If we went this direction, we might have `Removal.sol` be a base contract and then `SoilCarbonRemoval.sol` be the implementer which might start out as having little to no additional logic. Doing this would make it more obvious that internal things such as `tokenIdCounter`, `name`, etc should not be modified through anything other than via an explicit `override` function definition

Thoughts?

Reply:
also, we should use OZ's [counters](https://docs.openzeppelin.com/contracts/4.x/api/utils#Counters) for this